Improve documentation and rollout procedure (#123)
wuespace/telestion-project-daedalus2: application/README.md

The application part of a Telestion Project.

## Prerequisites

You will need Java 16 and Docker with `docker-compose` set up and ready to go.

> An text editor or IDE with auto-completion and gradle support is beneficial.

Reviewer: ```suggestion
A text editor or IDE with auto-completion and gradle support is beneficial.
```

## Setup

To fetch the dependencies locally you need a Github personal access token with `packages:read` rights.
Either add your github user name and the token as `GITHUB_ACTOR` and `GITHUB_TOKEN` to your environment variables 
or make a copy of the `gradle.properties.example` and name it `gradle.properties`.
Then insert your username and token into this file.

> No worry, this file should not be committed. (it's git ignored)

Reviewer 2: ```suggestion
No worries, this file should not be committed. (it's git ignored)
```

> or make a copy of `gradle.properties.example` and name it `gradle.properties`.
> Then insert your username and token into this file.
> No worry, this file should not be commited. (it's git ignored)
> 
> Finally resync gradle to download an set up the Telestion Core dependencies.

Reviewer: ```suggestion
or make a copy of the `gradle.properties.example` and name it `gradle.properties`.
Then insert your username and token into this file.
No worry, this file should not be committed. (it's git ignored)

Finally, resync gradle to download and set up the Telestion Core dependencies.
```

## Usage and Testing

### Native

Now you can run the application natively on your system.
Before you start, you need a MongoDB Server setup.
To start it, please type:

```
docker-compose --profile devel-native up -d
```

Now, the MongoDB server is running on port `27017`.
You can try to connect to the server if you installed the mongo-client-tools:

```
mongo
```

Let's start the application with gradle:

```shell
./gradlew run
```

or:

```
.\gradle.bat run
```

### In Docker

If you want to test your changes directly in docker, you can package it and run it beside the MongoDB container.

1. Please shutdown any running containers and services:

```
docker-compose --profile devel-native down
```

2. Create a distribution build with gradle:

```shell
./gradlew assembleDist
```

or:

```
.\gradle.bat assembleDist
```

3. Build the local image with `docker-compose`:

```
docker-compose --profile devel-docker build
```

4. And create and start the containers:

```
docker-compose --profile devel-docker up -d
```

That's it!

### Production mode

> If you want to run the application in production mode using the latest release of the project, then you need to simply call:

Reviewer 2: ```suggestion
If you want to run the application in production mode using the latest release of the project, then you simply need to call:
```
